src/main/java/co/cask/hydrator/plugin/batchSource/KafkaBatchSource.java

```diff
@@ -242,17 +242,17 @@ public Schema getMessageSchema() {
         throw new IllegalArgumentException(
           "Schema must contain at least one other field besides the time and key fields.");
       }
-      if (getKeyField() != null && !keyFieldExists) {
+      if (!getKeyField().equals("key") && !keyFieldExists) {
```

Comment:
I dont think you should put the first half condition, since if the schema does not have this field, the pipeline run will fail since in the transform we are setting this field in the `StructuredRecord`